### PR TITLE
Update machine-controller to v1.4.1 and add credentials validation

### PR DESCRIPTION
**Pull request by xmudrii**
**What this PR does / why we need it**:

This PR updates `machine-controller` to v1.4.1. The default root disk size in the Terraform script is increased to 50 GB.

**Does this PR introduce a user-facing change?**:
```release-note
Update machine-controller to v1.4.1
Add OS_TENANT_ID environment variable
```

/assign @kron4eg

### CODE REVIEW

**Comment by kron4eg** [comment]
@xmudrii but about about TenantID?

**Comment by xmudrii** [comment]
@kron4eg Fixed, PTAL and let me know if anything else is needed.

File: pkg/util/credentials/credentials.go [hunk 1]
```diff
@@ -101,6 +102,7 @@ func ProviderCredentials(p kubeone.CloudProviderName) (map[string]string, error)
 			{Name: "OS_PASSWORD"},
 			{Name: "OS_DOMAIN_NAME"},
 			{Name: "OS_REGION_NAME"},
+			{Name: "OS_TENANT_ID"},
```
Comment (kron4eg) [inline pkg/util/credentials/credentials.go]: The problem here is that only ether of two (`OS_TENANT_ID` or `OS_TENANT_NAME`) is required, but not both simultaneously. But this line makes both of them always required, which is not what MC expect.

Comment (xmudrii) [reply pkg/util/credentials/credentials.go]: Both are not required to be present, but what happens in such case? Should we throw an error or have higher priority for one of them?

Comment (kron4eg) [reply pkg/util/credentials/credentials.go]: That's the trick, we need to make sure that one of ether is present, we don't need both of them, [here's MC error check](https://github.com/kubermatic/machine-controller/pull/580/files#diff-e07452860a04f7019cbb8dfb86d357c1R353)

**Pull request merged**